### PR TITLE
imhex: unstable-2023-10-01 -> 1.32.2

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

Not compatible with LLVM 16.  We'll hopefully be bumping the default to 17 soon on Linux anyway, but since this package seems to be explicit about LLVM version required anyway.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/im/imhex/package.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { lib
 , stdenv
 , cmake
-, llvm
+, llvm_17
```
Comment (kirillrdy) [inline pkgs/by-name/im/imhex/package.nix:4]: should this be changed in all-packages.nix as per https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults ?

Comment (alyssais) [reply pkgs/by-name/im/imhex/package.nix]: Seems like it.  Changed!

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 286163` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>imhex</li>
  </ul>
</details>

**Review by cafkafk — CHANGES_REQUESTED** [review]
This seems to complain about a missing `bin/plugins/builtin.hexplug`, and fails to launch correctly on my machine.

**Comment by dunxen** [comment]
Also fails on `x86_64-linux` for me with "No Plugins" modal on startup.

<details>
<summary>Startup log</summary>

```
[22:07:23] [INFO]  [main]       Welcome to ImHex 1.32.2!
[22:07:23] [INFO]  [main]       Compiled using commit Unknown@Unknown
[22:07:23] [INFO]  [main]       Running on Linux 6.7.1 #1-NixOS SMP PREEMPT_DYNAMIC Sat Jan 20 10:53:02 UTC 2024 (x86_64)
[22:07:23] [INFO]  [main]       Native scaling set to: 1.0
[22:07:23] [INFO]  [main]       Using 'AMD' GPU
[22:07:23] [INFO]  [main]       Task 'Setting up environment' finished successfully in 0 ms
[22:07:23] [INFO]  [main]       Task 'Creating directories' finished successfully in 3 ms
[22:07:23] [INFO]  [main]       Task 'Loading settings' finished successfully in 0 ms
[22:07:23] [INFO]  [main]       Loading plugin 'ui.hexpluglib'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'ui.hexpluglib' failed: /nix/store/azsliqvri4lqzxvgqnj4802vhjci8mi7-imhex-1.32.2/lib/imhex/plugins/ui.hexpluglib: undefined symbol: _ZN3hex15ContentRegistry8Language15addLocalizationERKN8nlohmann16json_abi_v3_11_310basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS3_14adl_serializerES6_IhSaIhEEvEE!
[22:07:23] [INFO]  [main]       Loading plugin 'builtin.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'builtin.hexplug' failed: ui.hexpluglib: cannot open shared object file: No such file or directory!
[22:07:23] [INFO]  [main]       Loading plugin 'decompress.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'decompress.hexplug' failed: ui.hexpluglib: cannot open shared object file: No such file or directory!
[22:07:23] [INFO]  [main]       Loading plugin 'disassembler.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'disassembler.hexplug' failed: ui.hexpluglib: cannot open shared object file: No such file or directory!
[22:07:23] [INFO]  [main]       Loading plugin 'hashes.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'hashes.hexplug' failed: /nix/store/azsliqvri4lqzxvgqnj4802vhjci8mi7-imhex-1.32.2/lib/imhex/plugins/hashes.hexplug: undefined symbol: _ZN3hex15ContentRegistry8Language15addLocalizationERKN8nlohmann16json_abi_v3_11_310basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS3_14adl_serializerES6_IhSaIhEEvEE!
[22:07:23] [INFO]  [main]       Loading plugin 'script_loader.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'script_loader.hexplug' failed: ui.hexpluglib: cannot open shared object file: No such file or directory!
[22:07:23] [INFO]  [main]       Loading plugin 'visualizers.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'visualizers.hexplug' failed: /nix/store/azsliqvri4lqzxvgqnj4802vhjci8mi7-imhex-1.32.2/lib/imhex/plugins/visualizers.hexplug: undefined symbol: _ZN3hex15ContentRegistry8Language15addLocalizationERKN8nlohmann16json_abi_v3_11_310basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS3_14adl_serializerES6_IhSaIhEEvEE!
[22:07:23] [INFO]  [main]       Loading plugin 'yara_rules.hexplug'
[22:07:23] [ERROR] [libimhex]   Loading plugin 'yara_rules.hexplug' failed: ui.hexpluglib: cannot open shared object file: No such file or directory!
[22:07:23] [ERROR] [main]       No plugins found!
[22:07:23] [WARN]  [main]       Task 'Loading plugins' finished unsuccessfully in 5 ms
[22:07:23] [INFO]  [main]       ImHex fully started in 8ms
[22:07:23] [WARN]  [main]       All tasks finished, but some failed
[22:07:55] [INFO]  [main]       Exit task 'Saving settings' finished successfully
[22:07:55] [INFO]  [main]       Exit task 'Cleaning up shared data' finished successfully
[22:07:55] [INFO]  [main]       Exit task 'Unloading plugins' finished successfully
[22:07:55] [INFO]  [main]       Exit task 'Deleting old files' finished successfully
```

</details>

**Comment by Ninja3047** [comment]
I root caused this a bit ago
There was an issue in the pattern lang repo where it links a different version of the fmt library which causes the undefined symbol error.

I think this https://github.com/WerWolv/PatternLanguage/commit/1adcdd358d3772681242267ddd3459c9d0913796 and https://github.com/WerWolv/PatternLanguage/commit/ebdb630580a0fe71f061660abd3fcc7ba75da65d need to be applied

**Review by cafkafk — APPROVED** [review]
This seems to work now, LGTM :+1:

**Pull request merged**